### PR TITLE
fix(time-picker): Adding the disabled style for items which are outside the min/maxValue range

### LINKED ISSUE #3978: Define different styling for hours/minutes outside min and max values
## Is your feature request related to a problem? Please describe.
When there is a minValue and maxValue defined in the igx-time-picker, there is no indication and it is not possible to close the dialog, when trying to set a date that is outside the defined minimum or maximum.

## Describe the solution you'd like
The igx-time-picker should have some disabled styling for the hours/minutes that are outside the minValue/maxValue.

### PR DESCRIPTION
**Pull request by valadzhov**
Closes #3978 

### Additional information (check all that apply):
 - [X] Bug fix
 - [x] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [X] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by valadzhov** [comment]
Work on this PR will resume after 9.1.x

File: projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts [hunk 1]
```diff
@@ -257,6 +266,20 @@ export class IgxHourItemDirective {
...
+        if (this.minValueDate === undefined || this.maxValueDate === undefined) {
+            return false;
+        }
+        let hour = parseInt(this.value, 10);
+        if (this.selectedAmPm === 'PM') {
+            hour += 12;
+        }
+        const date = new Date(this.minValueDate);
+        date.setHours(hour);
+        return !(date <= this.maxValueDate && date >= this.minValueDate);
```
Comment (jackofdiamond5) [inline projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts:280]: Just for readability's sake.
```suggestion
        return date < this.minValueDate || this.maxValueDate < date;
```

#

Alternatevly, you could also use `greaterThanMinValue` and `lessThanMaxValue` methods from `DatePickerUtil`. They will spare you the need to check everything by hand:
```
return DatePickerUtil.lessThanMinValue(date, this.minValueDate)
    || DatePickerUtil.greaterThanMaxValue(date, this.maxValueDate);
```

File: projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts [hunk 2]
```diff
@@ -299,6 +334,22 @@ export class IgxMinuteItemDirective {
         return this.isSelectedMinute && this.itemList.isActive;
     }
 
+    @HostBinding('class.igx-time-picker__item--disabled')
+    get applyDisabledStyleForMinutes(): boolean {
```
Comment (jackofdiamond5) [inline projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts:338]: Same as `applyDisabledStyleForHours`.

File: projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts [hunk 3]
```diff
@@ -341,6 +407,24 @@ export class IgxSecondsItemDirective {
         return this.isSelectedSeconds && this.itemList.isActive;
     }
 
+    @HostBinding('class.igx-time-picker__item--disabled')
+    get applyDisabledStyleForSeconds(): boolean {
```
Comment (jackofdiamond5) [inline projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts:411]: Same as `applyDisabledStyleForHours`.

File: projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts [hunk 4]
```diff
@@ -383,6 +482,25 @@ export class IgxAmPmItemDirective {
         return this.isSelectedAmPm && this.itemList.isActive;
     }
 
+    @HostBinding('class.igx-time-picker__item--disabled')
+    get applyDisabledStyleForAmPm(): boolean {
```
Comment (jackofdiamond5) [inline projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts:486]: Same as  `applyDisabledStyleForHours`.

File: projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts [hunk 5]
```diff
@@ -578,6 +578,10 @@ export class IgxTimePickerComponent implements
      */
     public selectedAmPm: string;
 
+    public minValueConverted: Date;
+
+    public maxValueConverted: Date;
```
[the remark this answers is not in the document]
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts:583]: You are right, these variables weren't used anywhere. They are removed now.

File: projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts [hunk 6]
```diff
@@ -1267,7 +1271,10 @@ export class IgxTimePickerComponent implements
         return date;
     }
 
-    private _convertMinMaxValue(value: string): Date {
+    public convertMinMaxValue(value: string): Date {
+        if (value == null) {
```
[the remark this answers is not in the document]
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts:1275]: Done 👍

File: projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts [hunk 7]
```diff
@@ -1308,11 +1315,12 @@ export class IgxTimePickerComponent implements
 
         return date;
     }
+    }
```
[the remark this answers is not in the document]
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts:1318]: Done 👍

File: projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts [hunk 8]
```diff
@@ -257,6 +266,20 @@ export class IgxHourItemDirective {
         return this.isSelectedHour && this.itemList.isActive;
     }
 
+    @HostBinding('class.igx-time-picker__item--disabled')
+    get applyDisabledStyleForHours(): boolean {
+        if (this.minValueDate === undefined || this.maxValueDate === undefined) {
```
[the remark this answers is not in the document]
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts:271]: Done 👍

File: projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts (hunk 1 above)
Comment (valadzhov) [reply projects/igniteui-angular/src/lib/time-picker/time-picker.directives.ts:280]: I went with the two methods from the DatePickerUtil even though they are a bit more verbose.

**Pull request merged**